src/modules/tags/index.ts

```diff
@@ -300,17 +356,117 @@ export const tagsCreateEpic: MyEpic = (action$, state$, {client}) =>
         )
       )
     })
-  )
+  );
+
+  export const checkAndCreateTagsForAssetsEpic: MyEpic = (action$, state$, { client }) =>
+  action$.pipe(
+    filter(tagsSlice.actions.checkAndCreateTagsStart.match),
+    withLatestFrom(state$),
+    switchMap(([action, state]) => {
+      return client.observable.fetch<{ items: Asset[] }>(ASSET_QUERY).pipe(
+        mergeMap(result => {
+          const { items } = result;
+          const prepareTagOptionsActions$ = from(items).pipe(
+            concatMap(asset => {
+              if (asset.tags) {
+                const params = { assetId: asset._id, tags: asset.tags, currentAsset: asset };
+                return of(tagsSlice.actions.prepareTagOptions(params)).pipe(delay(10000));
+              }
+              return EMPTY;
+            }),
+          );
+
+          const finalAction$ = from(items).pipe(
+            toArray(),
+            mergeMap(processedItems => {
+              const allOperationsSuccessful = processedItems.length === items.length;
+              return of(
+                allOperationsSuccessful
+                  ? checkAndCreateTagsSuccess({ message: SUCCESS_MESSAGE })
+                  : checkAndCreateTagsFailure({ message: FAILURE_MESSAGE })
+              );
+            })
+          );
+
+          return concat(prepareTagOptionsActions$, finalAction$);
+        }),
+        catchError((error: any) =>
+          of(checkAndCreateTagsFailure({
+            message: error?.message || INTERNAL_ERROR_MESSAGE,
+            statusCode: error?.statusCode || 500
+          }))
+        )
+      );
+    })
+  );
+
+export const ensureTagsExistEpic: MyEpic = (action$, state$) =>
+  action$.pipe(
+    filter(tagsSlice.actions.prepareTagOptions.match),
+    mergeMap(action => {
+      let updatedTagOptions: any[] = [];
+      const existingTags = action.payload.currentAsset.opt?.media?.tags || [];
+      updatedTagOptions.push(...existingTags);
+
+      return from(action.payload.tags).pipe(
+        concatMap(tagName => {
+          const asset = action.payload.currentAsset;
+          const tagExists = Object.values(state$?.value?.tags?.byIds).some(tag => tag?.tag?.name?.current === tagName);
```

Comment:
Should be the proper name. Add  better name

```suggestion
          const isExists = Object.values(state$?.value?.tags?.byIds).some(tag => tag?.tag?.name?.current === tagName);
```